src/types.rs

```diff
@@ -1,11 +1,142 @@
+use crate::{Gain, IntegrationTime};
+
+/// Raw measurement data from the sensor.
 #[derive(Clone, Copy)]
 pub struct RawData {
     pub ch0_raw: u16,
     pub ch1_raw: u16,
 }
 
+/// A measurement from the sensor.
 #[derive(Clone, Copy)]
 pub struct LuxData {
-    pub lux_raw: RawData,
-    pub lux_phys: f32,
+    /// Raw sensor data.
+    pub(crate) lux_raw: RawData,
+
+    /// [`Gain`] this data was measured with.
+    pub(crate) gain: Gain,
+
+    /// [`IntegrationTime`] this data was measured with.
+    pub(crate) integration_time: IntegrationTime,
+}
+
+impl LuxData {
+    /// Raw value returned from the sensor
+    pub fn lux_raw(&self) -> &RawData {
+        &self.lux_raw
+    }
+
+    /// Value in Lux.
+    pub fn lux_phys(&self) -> f32 {
+        raw_to_lux(
+            self.lux_raw.ch1_raw,
+            self.lux_raw.ch0_raw,
+            self.gain,
+            self.integration_time,
+        )
+    }
+
+    /// Value in milli-Lux. Useful on platforms that do not have a floating
+    /// point unit.
+    pub fn lux_phys_u32(&self) -> u32 {
+        raw_to_lux_u32(
+            self.lux_raw.ch1_raw,
+            self.lux_raw.ch0_raw,
+            self.gain,
+            self.integration_time,
+        )
+    }
+}
+
+fn raw_to_lux(ch1_data: u16, ch0_data: u16, gain: Gain, itime: IntegrationTime) -> f32 {
+    let ratio = ch1_data as f32 / (ch0_data as f32 + ch1_data as f32);
+    let als_gain: f32 = gain.into();
+    let int_time: f32 = itime.into();
+
+    if ratio < 0.45 {
+        ((1.7743 * f32::from(ch0_data)) + (1.1059 * f32::from(ch1_data))) / als_gain / int_time
+    } else if (0.45..0.64).contains(&ratio) {
+        ((4.2785 * f32::from(ch0_data)) - (1.9548 * f32::from(ch1_data))) / als_gain / int_time
+    } else if (0.64..0.85).contains(&ratio) {
+        ((0.5926 * f32::from(ch0_data)) + (0.1185 * f32::from(ch1_data))) / als_gain / int_time
+    } else {
+        0.0
+    }
+}
+
+fn raw_to_lux_u32(ch1_data: u16, ch0_data: u16, gain: Gain, itime: IntegrationTime) -> u32 {
```

Comment:
`raw_to_millilux` would be more appropriate, don't you think?

Reply:
Yes, will change that.